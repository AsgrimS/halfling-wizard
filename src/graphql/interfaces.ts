import { Class } from "./schema";

export interface HeroClassesData {
  classes: Class[];
}

export interface HeroClassData {
  class: Class;
}
