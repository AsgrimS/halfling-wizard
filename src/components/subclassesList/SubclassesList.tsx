import React from "react";
import { SubclassesData } from "../../graphql/interfaces";

const SubclassesList: React.FC<SubclassesData> = ({ subclasses }) => {
  return (
    <div data-testid="subclasses-list">
      <span>Subclasses:</span>
      <ul>
        {subclasses.map((subclass) => (
          <li key={subclass.index}>{subclass.name}</li>
        ))}
      </ul>
    </div>
  );
};

export default SubclassesList;
