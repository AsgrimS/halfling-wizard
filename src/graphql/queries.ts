import { gql } from "@apollo/client";

export const HERO_CLASS = gql`
  query GetHeroesClasses($classIndex: String!) {
    class(filter: { name: $classIndex }) {
      hit_die
      index
      name
      proficiencies {
        index
        name
      }
      saving_throws {
        index
        name
      }
      proficiency_choices {
        choose
        from {
          name
          index
        }
      }
      spellcasting {
        level
        spellcasting_ability {
          index
          name
        }
      }
      starting_equipment {
        equipment {
          name
          index
        }
        quantity
      }
      starting_equipment_options {
        choose
        from {
          equipment {
            name
            index
          }
          quantity
        }
      }
      subclasses {
        index
        name
      }
    }
  }
`;

export const HEROES_CLASSES = gql`
  query GetHeroesClasses {
    classes {
      name
      index
    }
  }
`;
