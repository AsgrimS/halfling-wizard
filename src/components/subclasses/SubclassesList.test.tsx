import React from "react";
import { render, screen, cleanup } from "@testing-library/react";

import SubclassesList from "./SubclassesList";
import { Subclass } from "../../graphql/schema";

describe("SubclassesList component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render a list of subclasses", () => {
    const mockSubclassesListData: Subclass[] = [
      { name: "Warrior", index: "warrior" },
      { name: "Wizard", index: "wizard" },
    ];

    render(<SubclassesList subclasses={mockSubclassesListData} />);
    const SublcasessListElement = screen.getByTestId("subclasses-list");
    const liElements = SublcasessListElement.getElementsByTagName("li");

    expect(SublcasessListElement).toBeInTheDocument();
    expect(liElements.length).toEqual(2);

    for (let i = 0; i < liElements.length; i += 1) {
      expect(liElements[i]).toContainHTML(
        `<li>${mockSubclassesListData[i].name}</li>`
      );
    }

    expect(SublcasessListElement).toMatchSnapshot();
  });
});
