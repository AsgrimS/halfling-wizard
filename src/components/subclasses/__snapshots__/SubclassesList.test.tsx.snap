// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubclassesList component should render a list of subclasses 1`] = `
<div
  data-testid="subclasses-list"
>
  <span>
    Subclasses:
  </span>
  <ul>
    <li>
      Warrior
    </li>
    <li>
      Wizard
    </li>
  </ul>
</div>
`;
